engine/gameobject/src/gameobject/comp_script.cpp

```diff
@@ -379,6 +379,48 @@ namespace dmGameObject
                 lua_setfield(L, action_table, "gamepad_name");
             }
 
+            if (params.m_InputAction->m_HasGamepadPacket)
+            {
+                dmHID::GamepadPacket gamepadPacket = params.m_InputAction->m_GamepadPacket;
```

Comment:
I'm not sure about the performance difference here versus using a `GamepadPacker*`. On one hand we're copying the 144bytes, and have the data in a local cache, versus using a pointer, and it might thrash the caches.
I think this setup is probably better.

I only mention it since I don't think we normally use this pattern of copying a whole struct on the stack, but I'm thinking it's a good use case for itt.